online_party_system/party_match.go

```diff
@@ -129,19 +130,27 @@ func getBlockedFriendsWithCache(ctx context.Context, logger runtime.Logger, nk r
 	friendState := 3 // 3 is Friend_BLOCKED
 	cursor := ""     // no cursor
 
-	friendsPtrArray, cursor, err := nk.FriendsList(ctx, userID, friendLimit, &friendState, cursor)
-	if err != nil {
-		logger.Error("Error getting FriendsList: %v", err)
-		return nil
-	}
+	var blockedFriends []string
 
-	blockedFriends := make([]string, 0, len(friendsPtrArray))
+	for {
+		var friendsPtrArray []*api.Friend
+		var err error
+		friendsPtrArray, cursor, err = nk.FriendsList(ctx, userID, friendLimit, &friendState, cursor)
+		if err != nil {
+			logger.Error("Error getting FriendsList: %v", err)
+			return nil
+		}
+
+		for _, friendPtr := range friendsPtrArray {
+			friend := *friendPtr
+			user := friend.User
+			userID := user.Id
+			blockedFriends = append(blockedFriends, userID)
+		}
 
-	for _, friendPtr := range friendsPtrArray {
-		friend := *friendPtr
-		user := friend.User
-		userID := user.Id
-		blockedFriends = append(blockedFriends, userID)
+		if len(friendsPtrArray) < friendLimit {
```

Comment:
Make this check:
```
if len(friendsPtrArray) < friendLimit || cursor == "" {
```
Otherwise if there are less than `friendLimit` blocked users we'll loop over the same ones (cursor would always be `""`) and insert them into `blockedFriends` as duplicate IDs, or just loop forever if there is nobody blocked I think.

Reply:
Never mind, misread it originally. The check should just be `if cursor == ""`. An empty cursor is the standard way to check we've reached the end of our pagination.

Reply:
Thanks, that makes sense... will update